app/models/volunteer.rb

```diff
@@ -28,24 +28,35 @@ def name
 
   def self.import_csv(file)
     CSV.foreach(Rails.root.join(file), headers: true) do |row|
-      Volunteer.create({
+     Volunteer.create!({
         id: nil,
         first_name: row[0],
         last_name: row[1],
-        street_address:  row[2],
-        city: row[3],
-        state: row[4],
-        zip: row[5],
-        county: row[6],
-        phone: row[7],
+        street_address:  row[3],
+        city: row[4],
+        state: row[5],
+        zip: row[6],
+        county: row[7],
+        phone: row[8],
         university_location_id: nil, #depending on how locations are being created and used we could use Location.find(row[8].to_i).id
-        graduation_year: row[9],
-        ok_to_email: row[10],
-        ok_to_text: row[11],
-        ok_to_call: row[12],
-        ok_to_mail: row[13],
-        underage: row[14]
+        graduation_year: row[10],
+        ok_to_email: row[11],
+        ok_to_text: row[12],
+        ok_to_call: row[13],
+        ok_to_mail: row[14],
+        underage: row[15]
+      })
+      User.create({
+        id: nil,
+        email: row[2],
+        password: "Secret!!",
+        password_confirmation: "Secret!!",
+        volunteer_id: Volunteer.find_by(first_name: row[0], last_name: row[1]).id
       })
+
+      volunteer = Volunteer.find_by(first_name: row[0], last_name: row[1])
+      volunteer.user.email = User.find_by(volunteer_id: volunteer.id).email
```

Comment:
The user's email is set, but the volunteer's email relies on the user being created so that we can use the user's email. The volunteer create action does not use the email at first because we need to create the user and their email, but we cannot create the user first because we need the volunteer id first, where we find the volunteer by the csv first name and last name entries, does that make sense?

Reply:
@maebeale

Reply:
@maebeale Fixed!